File: test_nav_components.py
"""
DS Navigation Components - Interactive Testing Script
Tests: theme toggle, avatar dropdown, underline tabs, pagination
"""
from playwright.sync_api import sync_playwright
import time

FILE_URL = "file:///C:/scratchpad/pgs-redesign-2.0/.worktrees/ds-components/ds-s04-nav.html"
VIEWPORT = {"width": 1400, "height": 900}

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    page = browser.new_page(viewport=VIEWPORT)

    # ── Step 1: Full page screenshot (dark mode default) ──
    print("=== STEP 1: Full page screenshot (dark mode) ===")
    page.goto(FILE_URL)
    page.wait_for_load_state("networkidle")
    time.sleep(1)

    # Report page structure
    theme = page.get_attribute("html", "data-theme")
    print(f"  Theme: {theme}")

    # Check sidebar
    sidebar = page.locator(".ds-sidebar")
    sidebar_visible = sidebar.is_visible()
    print(f"  Sidebar visible: {sidebar_visible}")

    # Count sections
    sections = page.locator("main section").count()
    print(f"  Sections found: {sections}")

    # Section titles
    headings = page.locator(".ds-section__title").all_text_contents()
    print(f"  Section titles: {headings}")

    page.screenshot(path="/tmp/ds-nav-01-fullpage.png", full_page=True)
    print("  Screenshot saved: /tmp/ds-nav-01-fullpage.png")

    # ── Step 2: Click theme toggle to light mode ──
    print("\n=== STEP 2: Theme toggle → light mode ===")
    toggle = page.locator("#themeToggle")
    toggle_text_before = toggle.text_content()
    print(f"  Toggle icon before click: '{toggle_text_before.strip()}'")

    toggle.click()
    time.sleep(0.8)  # transition time

    theme_after = page.get_attribute("html", "data-theme")
    toggle_text_after = toggle.text_content()
    print(f"  Theme after click: {theme_after}")
    print(f"  Toggle icon after click: '{toggle_text_after.strip()}'")

    page.screenshot(path="/tmp/ds-nav-02-lightmode.png", full_page=True)
    print("  Screenshot saved: /tmp/ds-nav-02-lightmode.png")

    # ── Step 3: Click TM avatar dropdown ──
    print("\n=== STEP 3: TM avatar dropdown ===")
    # Switch back to dark mode first
    toggle.click()
    time.sleep(0.5)
    theme_back = page.get_attribute("html", "data-theme")
    print(f"  Switched back to: {theme_back}")

    # Scroll to the navbar section first
    page.locator("#navbar").scroll_into_view_if_needed()
    time.sleep(0.3)

    # Click the TM avatar
    avatar = page.locator(".nav-bar__avatar").first
    avatar_text = avatar.text_content()
    print(f"  Avatar text: '{avatar_text.strip()}'")
    avatar.click()
    time.sleep(0.5)

    # Check dropdown is open
    dropdown_open = page.locator(".nav-bar__user.is-open").first.is_visible()
    print(f"  Dropdown open: {dropdown_open}")

    # Read dropdown contents
    dropdown = page.locator("#userDropdown1 .nav-bar__dropdown")
    dropdown_name = dropdown.locator(".nav-bar__dropdown-name").text_content()
    dropdown_email = dropdown.locator(".nav-bar__dropdown-email").text_content()
    dropdown_items = dropdown.locator(".nav-bar__dropdown-item").all_text_contents()
    print(f"  User name: {dropdown_name.strip()}")
    print(f"  User email: {dropdown_email.strip()}")
    print(f"  Menu items: {[i.strip() for i in dropdown_items]}")

    page.screenshot(path="/tmp/ds-nav-03-avatar-dropdown.png", full_page=True)
    print("  Screenshot saved: /tmp/ds-nav-03-avatar-dropdown.png")

    # Close the dropdown by clicking elsewhere
    page.locator("body").click(position={"x": 10, "y": 10})
    time.sleep(0.3)

    # ── Step 4: Click Drafts tab ──
    print("\n=== STEP 4: Underline tabs → Drafts ===")
    # Scroll to tabs section
    page.locator("#tabs").scroll_into_view_if_needed()
    time.sleep(0.5)

    # Check which tab is currently active
    active_tab_before = page.locator('.tabs-underline[data-tabgroup="underline"] .tabs-underline__item.is-active').text_content()
    print(f"  Active tab before: '{active_tab_before.strip()}'")

    # Click Drafts tab
    drafts_tab = page.locator('[data-tabgroup="underline"] [data-tab="drafts"]')
    drafts_tab.click()
    time.sleep(0.5)

    # Check active tab changed
    active_tab_after = page.locator('.tabs-underline[data-tabgroup="underline"] .tabs-underline__item.is-active').text_content()
    print(f"  Active tab after: '{active_tab_after.strip()}'")

    # Read the drafts panel content
    drafts_panel = page.locator('[data-tabgroup="underline"][data-panel="drafts"]')
    drafts_visible = drafts_panel.is_visible()
    drafts_content = drafts_panel.text_content()
    print(f"  Drafts panel visible: {drafts_visible}")
    print(f"  Drafts panel content: '{drafts_content.strip()[:120]}...'")

    page.screenshot(path="/tmp/ds-nav-04-drafts-tab.png", full_page=True)
    print("  Screenshot saved: /tmp/ds-nav-04-drafts-tab.png")

    # ── Step 5: Click pagination number ──
    print("\n=== STEP 5: Pagination click ===")
    # Scroll to pagination section
    page.locator("#pagination").scroll_into_view_if_needed()
    time.sleep(0.5)

    # Find the active page before click
    active_page_before = page.locator("#paginationDemo .pagination__btn.is-active").text_content()
    print(f"  Active page before: '{active_page_before.strip()}'")

    # Click page 3
    page_buttons = page.locator("#paginationDemo .pagination__btn").all()
    print(f"  Total pagination buttons: {len(page_buttons)}")
    for btn in page_buttons:
        txt = btn.text_content().strip()
        if txt == "3":
            btn.click()
            print(f"  Clicked page: 3")
            break
    time.sleep(0.5)

    # Check active page changed
    active_page_after = page.locator("#paginationDemo .pagination__btn.is-active").text_content()
    print(f"  Active page after: '{active_page_after.strip()}'")

    # Check compact pagination text
    compact_texts = page.locator(".pagination-compact__text").all_text_contents()
    print(f"  Compact pagination texts: {[t.strip() for t in compact_texts]}")

    page.screenshot(path="/tmp/ds-nav-05-pagination.png", full_page=True)
    print("  Screenshot saved: /tmp/ds-nav-05-pagination.png")

    print("\n=== ALL STEPS COMPLETE ===")
    browser.close()

File: test_nav_components.js
/**
 * DS Navigation Components – Interactive Testing Script (Node.js / Playwright)
 * Tests: theme toggle, avatar dropdown, underline tabs, pagination
 */
const { chromium } = require('playwright');

const FILE_URL = 'file:///C:/scratchpad/pgs-redesign-2.0/.worktrees/ds-components/ds-s04-nav.html';
const VIEWPORT = { width: 1400, height: 900 };

(async () => {
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage({ viewport: VIEWPORT });

  // Helper: sleep
  const sleep = ms => new Promise(r => setTimeout(r, ms));

  // ── Step 1: Full page screenshot (dark mode default) ──────────────────
  console.log('=== STEP 1: Full page screenshot (dark mode) ===');
  await page.goto(FILE_URL);
  await page.waitForLoadState('networkidle');
  await sleep(1000);

  const theme = await page.getAttribute('html', 'data-theme');
  console.log('  Theme:', theme);

  const sidebarVisible = await page.locator('.ds-sidebar').isVisible();
  console.log('  Sidebar visible:', sidebarVisible);

  const sectionCount = await page.locator('main section').count();
  console.log('  Sections found:', sectionCount);

  const headings = await page.locator('.ds-section__title').allTextContents();
  console.log('  Section titles:', headings.map(h => h.trim()));

  await page.screenshot({ path: '/tmp/ds-nav-01-fullpage.png', fullPage: true });
  console.log('  Screenshot saved: /tmp/ds-nav-01-fullpage.png');

  // ── Step 2: Click theme toggle to light mode ──────────────────────────
  console.log('\n=== STEP 2: Theme toggle → light mode ===');
  const toggle = page.locator('#themeToggle');
  const iconBefore = (await toggle.textContent()).trim();
  console.log('  Toggle icon before:', JSON.stringify(iconBefore));

  await toggle.click();
  await sleep(800);

  const themeAfter = await page.getAttribute('html', 'data-theme');
  const iconAfter = (await toggle.textContent()).trim();
  console.log('  Theme after click:', themeAfter);
  console.log('  Toggle icon after:', JSON.stringify(iconAfter));

  await page.screenshot({ path: '/tmp/ds-nav-02-lightmode.png', fullPage: true });
  console.log('  Screenshot saved: /tmp/ds-nav-02-lightmode.png');

  // ── Step 3: Click TM avatar dropdown ──────────────────────────────────
  console.log('\n=== STEP 3: TM avatar dropdown ===');

  // Switch back to dark
  await toggle.click();
  await sleep(500);
  const themeBack = await page.getAttribute('html', 'data-theme');
  console.log('  Switched back to:', themeBack);

  // Scroll to navbar
  await page.locator('#navbar').scrollIntoViewIfNeeded();
  await sleep(300);

  // Click TM avatar
  const avatar = page.locator('.nav-bar__avatar').first();
  const avatarText = (await avatar.textContent()).trim();
  console.log('  Avatar text:', JSON.stringify(avatarText));
  await avatar.click();
  await sleep(500);

  // Check dropdown
  const dropdownOpen = await page.locator('.nav-bar__user.is-open').first().isVisible();
  console.log('  Dropdown open:', dropdownOpen);

  const dropdown = page.locator('#userDropdown1 .nav-bar__dropdown');
  const dropdownName = (await dropdown.locator('.nav-bar__dropdown-name').textContent()).trim();
  const dropdownEmail = (await dropdown.locator('.nav-bar__dropdown-email').textContent()).trim();
  const dropdownItems = await dropdown.locator('.nav-bar__dropdown-item').allTextContents();
  console.log('  User name:', dropdownName);
  console.log('  User email:', dropdownEmail);
  console.log('  Menu items:', dropdownItems.map(i => i.trim()));

  await page.screenshot({ path: '/tmp/ds-nav-03-avatar-dropdown.png', fullPage: true });
  console.log('  Screenshot saved: /tmp/ds-nav-03-avatar-dropdown.png');

  // Close dropdown
  await page.locator('body').click({ position: { x: 10, y: 10 } });
  await sleep(300);

  // ── Step 4: Click Drafts tab ──────────────────────────────────────────
  console.log('\n=== STEP 4: Underline tabs → Drafts ===');

  await page.locator('#tabs').scrollIntoViewIfNeeded();
  await sleep(500);

  const activeTabBefore = (await page.locator('.tabs-underline[data-tabgroup="underline"] .tabs-underline__item.is-active').textContent()).trim();
  console.log('  Active tab before:', JSON.stringify(activeTabBefore));

  const draftsTab = page.locator('[data-tabgroup="underline"] [data-tab="drafts"]');
  await draftsTab.click();
  await sleep(500);

  const activeTabAfter = (await page.locator('.tabs-underline[data-tabgroup="underline"] .tabs-underline__item.is-active').textContent()).trim();
  console.log('  Active tab after:', JSON.stringify(activeTabAfter));

  const draftsPanel = page.locator('[data-tabgroup="underline"][data-panel="drafts"]');
  const draftsVisible = await draftsPanel.isVisible();
  const draftsContent = (await draftsPanel.textContent()).trim();
  console.log('  Drafts panel visible:', draftsVisible);
  console.log('  Drafts panel content:', JSON.stringify(draftsContent.substring(0, 150)));

  await page.screenshot({ path: '/tmp/ds-nav-04-drafts-tab.png', fullPage: true });
  console.log('  Screenshot saved: /tmp/ds-nav-04-drafts-tab.png');

  // ── Step 5: Click pagination number ───────────────────────────────────
  console.log('\n=== STEP 5: Pagination click ===');

  await page.locator('#pagination').scrollIntoViewIfNeeded();
  await sleep(500);

  const activePageBefore = (await page.locator('#paginationDemo .pagination__btn.is-active').textContent()).trim();
  console.log('  Active page before:', activePageBefore);

  const pageButtons = await page.locator('#paginationDemo .pagination__btn').all();
  console.log('  Total pagination buttons:', pageButtons.length);

  for (const btn of pageButtons) {
    const txt = (await btn.textContent()).trim();
    if (txt === '3') {
      await btn.click();
      console.log('  Clicked page: 3');
      break;
    }
  }
  await sleep(500);

  const activePageAfter = (await page.locator('#paginationDemo .pagination__btn.is-active').textContent()).trim();
  console.log('  Active page after:', activePageAfter);

  const compactTexts = await page.locator('.pagination-compact__text').allTextContents();
  console.log('  Compact pagination texts:', compactTexts.map(t => t.trim()));

  await page.screenshot({ path: '/tmp/ds-nav-05-pagination.png', fullPage: true });
  console.log('  Screenshot saved: /tmp/ds-nav-05-pagination.png');

  console.log('\n=== ALL STEPS COMPLETE ===');
  await browser.close();
})();
